go_pprof_scraper/datadog_checks/go_pprof_scraper/check.py

```diff
@@ -0,0 +1,208 @@
+        # TODO: for some reason the config I get here doesn't match the real
+        # config when I run "ddev env start --dev go_pprof_scraper py38". Here,
+        # I see apm_config with enabled: True, the default receiver_port, etc.
+        # But in the agent logs ("docker logs dd_go_pprof_scraper_py38") I see
+        # stuff like
+        #
+        #   [date] TRACE | INFO | (run.go:103 in Run) | trace-agent not enabled. Set the environment variable
+        #   DD_APM_ENABLED=true or add "apm_config.enabled: true" entry
+        #   to your datadog.yaml. Exiting...
+        apm_config = datadog_agent.get_config("apm_config")
+        if not apm_config:
+            self.apm_enabled = False
+            return
+
+        self.apm_enabled = apm_config["enabled"]
+        self.trace_agent_port = apm_config["receiver_port"]
+        # XXX: Will the hostname ever *not* be localhost? I don't think so since
+        # this is being run by the agent
+        self.trace_agent_url = "http://localhost:{}/profiling/v1/input".format(self.trace_agent_port)
+
+        self.trace_agent_socket = quote(apm_config.get("receiver_socket", "").encode())
+        if self.trace_agent_socket:
+            self.trace_agent_socket = "http+unix://{}/profiling/v1/input".format(self.trace_agent_socket)
+
+    def _get_profile(self, profile):
+        query_params = {}
+        if profile != "cpu" and self.cumulative:
+            # cumulative profiles should still be collected *after*
+            # self.duration so that all of the returned profiles reflect the
+            # change in program state over the same time period
+            time.sleep(self.duration)
+        else:
+            query_params["seconds"] = self.duration
+        url = urljoin(self.url, "profile" if profile == "cpu" else profile)
+
+        response = self.http.get(
+            url,
+            params=query_params,
+            # We want to stream the response so that we can forward it to the
+            # upload, ideally without having to actually read the whole profile
+            # into memory
+            stream=True,
+            # We can get bit by the default timeout here when passing the "seconds"
+            # query parameter. By design, the request takes that long to finish.
+            #
+            # requests supports a timeout parameter, which can be a tuple
+            # (connect_timeout, read_timeout). The connect_timeout can be short, but
+            # the read_timeout needs to be longer than the profile duration.
+            timeout=(3.05, self.duration + 3),
+        )
+
+        response.raise_for_status()
+        if profile == "cpu" or self.cumulative:
+            name = profile
+        else:
+            name = "delta-{}".format(profile)
+        return name, response
+
+    def check(self, _):
+        if not self.apm_enabled:
+            self.service_check("can_connect", AgentCheck.CRITICAL, message="the trace agent is not enabled")
```

Comment:
This check should be declared in the `service_checks.json` file

Reply:
Done. What should I set the "agent_version" field to, though? The latest one? I just copied another file for now.

Reply:
7.40 should be ok because I'm not sure 7.41 will be out soon enough

Reply:
Got it, updated